sycl/include/sycl/info/ext_codeplay_device_traits.def

```diff
@@ -3,6 +3,9 @@
 #define __SYCL_PARAM_TRAITS_TEMPLATE_SPEC __SYCL_PARAM_TRAITS_SPEC
 #endif
 __SYCL_PARAM_TRAITS_SPEC(ext::codeplay::experimental,device, supports_fusion, bool, PI_EXT_CODEPLAY_DEVICE_INFO_SUPPORTS_FUSION)
+__SYCL_PARAM_TRAITS_SPEC(
+    ext::codeplay::experimental, device, max_registers_per_work_group, uint32_t,
```

Comment:
Where is this extension documented?

Reply:
I've now added `sycl/doc/extensions/experimental/sycl_ext_codeplay_max_registers_per_work_group_query.md`. I have seen another extension for `num_regs` having just an inline documentation where it was added in the PI header, but thought I'd add a full doc for `sycl_ext_codeplay_max_registers_per_work_group` with a MACRO.
Does that sound okay to you? And if so, and I am missing something, please let me know. Thank you! :)

Reply:
Thank you! Yes, I am okay with that and I think I see the one you are referring to. That one might have slipped through the cracks, but the general rule is that public interfaces like that should either be specification or extension interfaces.

@AerialMantis - On a related note, are there any particular rhyme or reason to when we use `codeplay` in the extension features vs. `intel` and/or `oneapi`?